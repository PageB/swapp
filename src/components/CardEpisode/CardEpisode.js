import React from 'react';
import styles from './CardEpisode.module.css';

const cardEpisode = props => {
  return (
    <div className={styles.Card} onClick={() => props.navTo(props.card.id)}>
      <img className={styles.CardImage} src={props.card.image} alt={props.card.title} />
      <div className={styles.CardDetails}>
        <p className={styles.CardTitle}>{props.card.title}</p>
        <p className={styles.CardSummary}>{props.card.summary}</p>
      </div>
    </div>
  );
};

export default cardEpisode;
