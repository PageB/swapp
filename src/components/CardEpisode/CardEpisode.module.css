.Card {
  background-color: #333333;
  border: 1px solid #3c4858;
  border-radius: 8px;
  color: #ffe300;
  width: 280px;
  cursor: pointer;
}

.CardImage {
  border-radius: 8px 8px 0px 0px;
  width: 280px;
}

.CardDetails {
  color: inherit;
  background-color: inherit;
  padding: 0 16px;
}

.CardTitle {
  border-radius: 8px 8px 0px 0px;
  color: inherit;
  font-weight: bold;
  font-size: 22px;
}

.CardSummary {
  color: #abb1ba;
  font-weight: normal;
  display: block;
  display: -webkit-box;
  -webkit-line-clamp: 8;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
}
