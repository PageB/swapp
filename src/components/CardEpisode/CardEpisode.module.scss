@import '../../colors.scss';
@import '../../constants.scss';

.CardEpisode {
  background-color: $dark-charcoal;
  color: $vivid-yellow;
  width: $card-width;
  margin: 8px 16px;
  border-radius: 8px;
  cursor: pointer;

  &__Image {
    border-radius: 8px 8px 0 0;
    width: 100%;
  }

  &__Details {
    color: inherit;
    background-color: inherit;
    padding: 0 16px;
  }

  &__Title {
    border-radius: 8px 8px 0 0;
    color: inherit;
    font-weight: bold;
    font-size: 22px;
  }

  &__Summary {
    color: $cadet-blue;
    font-weight: normal;
    display: block;
    display: -webkit-box;
    -webkit-line-clamp: 8;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}

.DarkTheme {
  background-color: $dark-charcoal;
  color: $vivid-yellow;
}

.LightTheme {
  background-color: $white;
  color: $crayola;
}
