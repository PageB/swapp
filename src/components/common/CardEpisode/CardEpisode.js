import React from 'react';
import styles from './CardEpisode.module.css';
import { withRouter } from 'react-router-dom';

const cardEpisode = withRouter(props => {
  return (
    <div
      className={[styles.Card, styles[props.theme]].join(' ')}
      onClick={() => {
        props.history.push(`/episode/${props.card.id}`, props.card);
      }}>
      <img className={styles.CardImage} src={props.card.image} alt={props.card.title} />
      <div className={styles.CardDetails}>
        <p className={styles.CardTitle}>{props.card.title}</p>
        <p className={styles.CardSummary}>{props.card.summary}</p>
      </div>
    </div>
  );
});

export default cardEpisode;
