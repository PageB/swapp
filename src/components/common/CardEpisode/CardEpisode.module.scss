@import '../../../colors.scss';

.Card {
  background-color: $dark-charcoal;
  border-radius: 8px;
  color: $vivid-yellow;
  width: 280px;
  margin: 8px 16px;
  cursor: pointer;
}

.CardImage {
  border-radius: 8px 8px 0px 0px;
  width: 280px;
}

.CardDetails {
  color: inherit;
  background-color: inherit;
  padding: 0 16px;
}

.CardTitle {
  border-radius: 8px 8px 0px 0px;
  color: inherit;
  font-weight: bold;
  font-size: 22px;
}

.CardSummary {
  color: $cadet-blue;
  font-weight: normal;
  display: block;
  display: -webkit-box;
  -webkit-line-clamp: 8;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
}

.DarkTheme {
  background-color: $dark-charcoal;
  color: $vivid-yellow;
}

.LightTheme {
  background-color: $white;
  color: $crayola;
}
