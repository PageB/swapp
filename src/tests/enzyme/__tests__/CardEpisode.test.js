import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import { shallow, configure } from 'enzyme';
import { starWarsEpisodes } from '../../../mock/data';

import CardEpisode from '../../../components/CardEpisode/CardEpisode';

configure({ adapter: new Adapter() });

describe('CardEpisode Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CardEpisode card={starWarsEpisodes[0]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Calls closure when card is clicked', () => {
    const onCardClick = sinon.spy();
    const wrapper = shallow(<CardEpisode card={starWarsEpisodes[0]} navigate={onCardClick} />);

    wrapper.find('.CardEpisode').simulate('click');

    expect(onCardClick.calledOnce).toBe(true);
  });
});
