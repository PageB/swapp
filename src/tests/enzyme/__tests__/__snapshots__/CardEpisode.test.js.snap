// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardEpisode Component should match snapshot 1`] = `
<div
  className="CardEpisode DarkTheme"
  onClick={[Function]}
>
  <img
    alt="The Phantom Menace"
    className="CardEpisode__Image"
    src="https://images-na.ssl-images-amazon.com/images/I/81Vl2RQrZkL._SX522_.jpg"
  />
  <div
    className="CardEpisode__Details"
  >
    <p
      className="CardEpisode__Title"
    >
      The Phantom Menace
    </p>
    <p
      className="CardEpisode__Summary"
    >
      The Trade Federation upsets order and disrupts peace in the Galactic Republic by blockading the planet Naboo in preparation for a full-scale invasion. The Republic's leader, Supreme Chancellor Valorum, dispatches Jedi Master Qui-Gon Jinn and his apprentice, Obi-Wan Kenobi, to negotiate with Trade Federation Viceroy Nute Gunray. Darth Sidious, a Sith Lord and the Trade Federation's secret benefactor, orders the Viceroy to kill the Jedi and begin their invasion with an army of battle droids. The Jedi escape and flee to Naboo. During the invasion, Qui-Gon saves the life of a clumsy Gungan outcast, Jar Jar Binks, from being run over by a droid transport. Indebted to Qui-Gon, Jar Jar leads the Jedi to Otoh Gunga, an underwater city of Naboo. The Jedi try to persuade the Gungan leader, Boss Nass, to help the planet's surface dwellers, but are unsuccessful. However, the Jedi manage to obtain Jar Jar's guidance and underwater transport to Theed, the capital city of Naboo; they narrowly avoid getting eaten by sea monsters on the way. They rescue Naboo's queen, Padm√© Amidala, and escape from the blockaded planet on her Royal Starship, intending to reach the Republic capital planet of Coruscant.
    </p>
  </div>
</div>
`;
